### PR TITLE
feat(starfish) Add release information to span summary panel

### PR DESCRIPTION
**Pull request by gggritso**
Copymore goodies from the database panel!

**e.g.,**
Screenshot 2023-05-19 at 2 23 56 PM

### CODE REVIEW

File: static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    WHERE group_id = '${span.group_id}'
+    ORDER BY start_timestamp ASC
+    LIMIT 1
+ `
+    : null;
+
+  const result = useQuery<(Span & HasTransaction)[]>({
+    queryKey: ['span', 'first', span?.group_id],
+    queryFn: () =>
+      fetch(`${HOST}/?query=${query}&referrer=${referrer}`).then(res => res.json()),
```
Comment (DominikB2014) [inline static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx:24]: Thoughts on something like this
```suggestion
      fetch(`${HOST}/?query=${query}&referrer=${referrer}-first-seen-span`).then(res => res.json()),
```
And referer we would pass in `event-panel`.
That way you'll always know where the query is defined and where it's being called.

File: static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import {useQuery} from 'sentry/utils/queryClient';
+import {useQueryGetEvent} from 'sentry/views/starfish/modules/databaseModule/queries';
+import {HOST} from 'sentry/views/starfish/utils/constants';
+import type {Span} from 'sentry/views/starfish/views/spans/spanSummaryPanel/types';
+
+type HasTransaction = {
+  transaction_id: string;
+};
+
+function useFirstSeenSpan(span?: Span, referrer = 'first-seen-span') {
```
Comment (DominikB2014) [inline static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx:10]: Small nits, but for consisntency we've been prefixing query specific hooks with `useQuery`
We haven't actually standardized this so up to you to resolve

Also, I like the idea of passing in what the function needs vs what you expect to pass into it. Seems a bit more generic, say you only have the group_id.
```suggestion
function useQueryFirstSeenSpan(span?: Pick<Span, 'group_id'>;, referrer = 'first-seen-span') {
```
Would also need to update the other one below.

File: static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const result = useQuery<(Span & HasTransaction)[]>({
+    queryKey: ['span', 'last', span?.group_id],
+    queryFn: () =>
+      fetch(`${HOST}/?query=${query}&referrer=${referrer}`).then(res => res.json()),
+    retry: false,
+    initialData: [],
+    enabled: Boolean(span),
+  });
+
+  return {...result, data: result.data[0]};
```
Comment (DominikB2014) [inline static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx:61]: I believe we loose typing by doing this?
We could cast is
```suggestion
  return {...result, data: result.data[0]} as DefinedUseQueryResult<Span & HasTransaction>
```

**Review by DominikB2014 — APPROVED** [review]
Nice 👍 just left some nits

**Review by DominikB2014 — APPROVED** [review]
Nice 👍 just left some nits

File: static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx (hunk 3 above)
Comment (gggritso) [reply static/app/views/starfish/views/spans/spanSummaryPanel/useSpanSeenEvent.tsx:61]: Nope, typing is preserved, good Q though!

**Comment by gggritso** [comment]
@DominikB2014 fair points! I'll put 'em in my back pocket for now, but I'll come back to standardizing how queries are ran hopefully soon-ish!

**Comment by DominikB2014** [comment]
> @DominikB2014 fair points! I'll put 'em in my back pocket for now, but I'll come back to standardizing how queries are ran hopefully soon-ish!

@gggritso cool, loop me into any ideas, would be interested!

**Pull request merged**